### PR TITLE
Controller Support Update

### PR DESCRIPTION
**Pull request by derpmastear**
yes yes yes

### CODE REVIEW

File: frontend/dashboard/dashboard.py [hunk 1]
```diff
@@ -40,6 +42,11 @@ def update_widget(self, widget_name: str, data: dict):
             return
 
         self.widgets[widget_name].update(data)
+    #for controller
+    def update_widget(self, widget_name: str):
```
Comment (sparthaje) [inline frontend/dashboard/dashboard.py]: this method should also take in a dictionary of data to pass to the widget

File: frontend/dashboard/widget/controller.py [hunk 2]
```diff
@@ -0,0 +1,379 @@
...
+
+    def on_missed_packet(self, number):
+        pass
+#Definitions for the joystick methods
+list(map(XInputJoystick.register_event_type, [
+    'on_state_changed',
+    'on_axis',
+    'on_button',
+    'on_missed_packet',
+]))
```
Comment (sparthaje) [inline frontend/dashboard/widget/controller.py]: The widget should only have stuff to update an image of a controller to show what buttons are being pressed or something. 

Can you move all controller code into `p2s/manipulator/XInputJoystick.py` and `p2s/manipulator/manipulator.py`.

File: frontend/frontend.py [hunk 3]
```diff
@@ -10,4 +10,5 @@ def __init__(self):
     def start(self):
         while True:
             self.dashboard.update()
+            self.dashboard.update_widget("ControllerWidget")
```
Comment (sparthaje) [inline frontend/frontend.py]: `p2s/manipulator` module will call this command, so we can delete this line.

File: frontend/dashboard/widget/controller.py (hunk 2 above)
Comment (sparthaje) [reply frontend/dashboard/widget/controller.py]: XInput looks tinteresting. 

@davidxiang101 is also working on controller code rn, so you can talk to him on slack about it.

Comment (derpmastear) [reply frontend/dashboard/widget/controller.py]: lol it works

File: frontend/frontend.py (hunk 3 above)
Comment (derpmastear) [reply frontend/frontend.py]: o k

**Pull request closed without merging**